### ISSUE TITLE
Custom Armor guide leads to stackable, non-damageable armor items

### ISSUE THREAD

**Issue by DWalz** [FabricMC/fabric-docs#151]
**Docs version:** 1.21

If you follow the guide to create the Guidite Armor, the resulting items have no durability, which makes them stackable to 64 and take no damage when the player is hit. The item initializer uses the default-constructed `Item.Settings()` which do not include a `maxDamage` modifier. 

I believe in general that the durability for armor should be explained better, because it is quite unintuitive in the regard that you need a custom armor material which, unlike its `ToolMaterial` counterpart does not have any durability information. Therefor the durability of armor items has to be custom-set and is usually (e.g. vanilla items) derived from a general "material durability" and a specific "slot durability" for each armor slot. 

I think it would be good to have a small section that explains why you need to manually set the durability and some best-practices on calculating the value (material durability * slot durability).

**Comment by IMB11** [comment]
The ArmorItem constructor should apply properties to the provided Item.Settings based on the material used?

**Comment by IMB11** [comment]
The reference mod does not have stackable armor items, not sure if you've misread the page or misunderstood something.

**Comment by DWalz** [comment]
I don't know if I misread anything but I can reproduce the issue using the `latest` project of the reference mod. I also do not see how the `ArmorItem` can have any clue about how durable the item is, since the armor material is no longer created by implementing `ArmorMaterial` which was an interface on last version. It is now a record, hence we register it. We no longer implement `ArmorMaterial`, even in the reference mod. 

In the 1.20.4 version of the docs and the reference mod it still works because of `ArmorMaterial#getDurability` but the new `ArmorMaterial` record in 1.21 does not store any durability information.

**Comment by IMB11** [comment]
Seems like it changed during a prerelease of 1.21, that's annoying, someone will have to make a PR to fix this as I'm unable to at the moment.

**Comment by DWalz** [comment]
I can do that if you want to.

The only problem/question I'd have is how to store the durability information. In my own version of the docs mod I stored it as a constant alongside the registered armor material, but I don't know if that is best practice. 😄 

```java
public static final int GUIDITE_DURABILITY_MODIFIER = 15;
public static final RegistryEntry<ArmorMaterial> GUIDITE = ...
```

**Comment by IMB11** [comment]
can you not just put it into the item.settings?

**Comment by IMB11** [comment]
with a comment accompanying it?

**Comment by DWalz** [comment]
The reason I did that is because the durability is (usually) dependent on the armor slot that it is equipped in and then multiplied by the common "armor material durability" multiplier - or at least that is how the vanilla armor items are registered. I've also seen that in other mods I researched for that purpose.

My registration of the helmet for example is the following:

```java
public static final Item GUIDITE_HELMET = register(new ArmorItem(ModArmorMaterials.GUIDITE, ArmorItem.Type.HELMET, new Item.Settings().maxDamage(ArmorItem.Type.HELMET.getMaxDamage(ModArmorMaterials.GUIDITE_DURABILITY_MODIFIER))), "guidite_helmet");
```

It derives its `maxDamage` from `ArmorItem.Type.HELMET`, which has a `baseMaxDamage` specified by Minecraft itself:

```java
HELMET(EquipmentSlot.HEAD, 11, "helmet")
```

The same goes for the other armor slots. I found it to be more concise to set it as a constant and then use the same constant for every armor item instead of writing `15` in the `getMaxDamage` function but I can also put it in there directly.

Hope this is understandable and thank you for taking the time!
